lib/pbench/cli/server/report.py

```diff
@@ -242,36 +265,86 @@ def report_cache(tree: CacheManager):
                 if referenced > newest_cache:
                     newest_cache = referenced
                     newest_cache_name = tarball.name
-            cached_count += 1
-            size = Metadata.getvalue(tarball.dataset, Metadata.SERVER_UNPACKED)
-            if not size:
-                detailer.error(f"{tarball.name} has no unpacked size")
-                lacks_size += 1
-            elif not isinstance(size, int):
-                detailer.error(
-                    f"{tarball.name} has non-integer unpacked size "
-                    f"{size!r} ({type(size)})"
-                )
-                bad_size += 1
-            else:
-                if size < smallest_cache:
-                    smallest_cache = size
-                    smallest_cache_name = tarball.name
-                if size > biggest_cache:
-                    biggest_cache = size
-                    biggest_cache_name = tarball.name
-                cached_size += size
+            if tarball.unpacked:
+                cached_count += 1
+        size = Metadata.getvalue(tarball.dataset, Metadata.SERVER_UNPACKED)
+        if not size:
+            detailer.error(f"{tarball.name} has no unpacked size")
+            lacks_size += 1
+        elif not isinstance(size, int):
+            detailer.error(
+                f"{tarball.name} has non-integer unpacked size "
+                f"{size!r} ({type(size).__name__})"
+            )
+            bad_size += 1
+        else:
+            if size < smallest_cache:
+                smallest_cache = size
+                smallest_cache_name = tarball.name
+            if size > biggest_cache:
+                biggest_cache = size
+                biggest_cache_name = tarball.name
+            cached_size += size
+
+            # Check compression ratios
+            tar_size = tarball.tarball_path.stat().st_size
+            ratio = float(size - tar_size) / float(size)
+            if ratio < smallest_compression:
+                smallest_compression = ratio
+                smallest_compression_name = tarball.name
```

Comment:
Given that we tend to associate a name with a given stat, consider using a collection/object to tie these two items together (Python won't worry too much about the data type of the stat unless you do...).

If you go with an object, you could encapsulate the condition into the setter operation.  And, if the object knows whether it's a maximiser or a minimizer, then you could just have a single setter function (or, the object could track both the max and the min using a single entrypoint).  And, the object initializer could take care of managing the initial value.

Reply:
Yep:  it's overkill for one metric -- ditto for "one more" metric -- it's only when you look back and find that you've done 30 or so, including 5 sets of max/min/with-names, that it starts to look reasonable...but, by then, everything's already done!